Clustering/src/KMeans.java

```diff
@@ -1,10 +1,117 @@
 /**
  * K-Means clustering
  */
+import java.lang.Math;
+import java.util.ArrayList;
+
 public class KMeans  extends Clustering{
 
+    /**
+     * Clusters data via the K-means Algorithm
+     */
     @Override
     public int[] cluster(double[][] data, int numClusters){
+        int cycles =0 ;
+        double[][] centroids = initCentroids(data, numClusters);
+        while(!hasConverged(cycles)){
+            ArrayList<Integer> labels = getLabels(centroids, data);
+            centroids = getCentroids(labels, data, numClusters);
+            cycles++;
+        }
         return null;
```

Comment:
This method needs to return a list of integers. The list has an entry for each data point in the set, the value in any index represents which cluster that data point is in. For example, with 3 clusters, the only values in this list would be 0, 1 and 2.   
I'm not sure if that's how the labels list is organized or not, you may have to convert it.

Reply:
done